### ISSUE TITLE
API: downcating on fillna no-ops intentional?

### ISSUE THREAD

**Issue by jbrockmendel** [pandas-dev/pandas#45423]
```python
ser = pd.Series([1, 2, 3], dtype=object)

>>> ser.fillna(4).dtype
dtype('int64')
```

Better to retain dtype in these cases?

**Comment by jreback** [comment]
this was the point of the downcast kw

as we normally will cast object if possible

**Comment by jorisvandenbossche** [comment]
IMO we shouldn't retain or not retain dtype based on whether it was in practice a no-op or not. In which case the discussion is about downcasting in general, like:

```
In [31]: ser = pd.Series([1, 2, 3, np.nan], dtype=object)

In [32]: ser.fillna(4)
Out[32]: 
0    1
1    2
2    3
3    4
dtype: int64
```

**Comment by jbrockmendel** [comment]
closed by #45673
